Source/WebCore/animation/ScrollTimeline.cpp

```diff
@@ -138,4 +140,40 @@ AnimationTimeline::ShouldUpdateAnimationsAndSendEvents ScrollTimeline::documentW
     return AnimationTimeline::ShouldUpdateAnimationsAndSendEvents::No;
 }
 
+ScrollableArea* ScrollTimeline::scrollableAreaForSourceRenderer(RenderElement* renderer, Ref<Document> document)
+{
+    CheckedPtr renderBox = dynamicDowncast<RenderBox>(renderer);
+    if (!renderBox)
+        return nullptr;
+
+    if (renderer->element() == document->documentElement())
+        return &renderer->view().frameView();
+    return (renderBox->canBeScrolledAndHasScrollableArea() && renderBox->layer()) ? renderBox->layer()->scrollableArea() : nullptr;
+}
+
+ScrollTimeline::Data ScrollTimeline::computeScrollTimelineData() const
+{
+    if (!m_source)
+        return { };
+
+    auto* sourceScrollableArea = scrollableAreaForSourceRenderer(m_source->renderer(), m_source->document());
+    if (!sourceScrollableArea)
+        return { };
```

Comment:
Should this assert? Can we get here with a ScrollTimeline having a renderer that is not scrollable?

Reply:
Its incorrect to give a non-scrollable element as a source but we shouldn't assert.